namui-cli/electron/src/main.js

```diff
@@ -1,17 +1,41 @@
 const { app, BrowserWindow, ipcMain } = require("electron");
 const isDev = require("electron-is-dev");
+const { existsSync, readFileSync } = require("fs");
 const path = require("path");
 
 const config = getConfig();
+const gotTheLock = app.requestSingleInstanceLock();
+
+let mainWindow;
+
+if (!gotTheLock) {
+    app.quit();
+} else {
+    setAppAsDefaultProtocolClient(app, config.deepLinkSchemes);
+
+    app.whenReady().then(() => {
+        ipcMain.handle("config", () => config);
+        createWindow();
+    });
+
+    app.on("activate", function () {
+        if (BrowserWindow.getAllWindows().length === 0) createWindow();
+    });
+
+    app.on("window-all-closed", function () {
+        if (process.platform !== "darwin") app.quit();
+    });
+}
 
 function createWindow() {
-    const mainWindow = new BrowserWindow({
+    mainWindow = new BrowserWindow({
```

Comment:
why const removed?

Reply:
It's weird...
It doesn't seem to have to be a global variable. I think I tried it during prototyping. I will revert it.
https://github.com/NamseEnt/namseent/pull/294/files/6bd08fa77e244bc5017495252b8a3d0b33d143a1#diff-bd152e97aedadf849d265dabab4438788196f4db0395d988755a138c5476ee9cR9